Realm/RLMMongoCollection.mm

```diff
@@ -0,0 +1,307 @@
+                  completion:(RLMMongoFindOneBlock)completion {
+    [self findOneDocumentWhere:document options:[[RLMFindOptions alloc] init] completion:completion];
+}
+
+- (void)insertOneDocument:(NSDictionary<NSString *, id<RLMBSON>> *)document
+               completion:(RLMMongoInsertBlock)completion {
+    [self collection:self.name].insert_one(static_cast<realm::bson::BsonDocument>(RLMConvertRLMBSONToBson(document)),
+                                           [completion](realm::util::Optional<realm::ObjectId> objectId,
+                                                        realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        completion([[RLMObjectId alloc] initWithValue:*objectId], nil);
+    });
+}
+
+- (void)insertManyDocuments:(NSArray<NSDictionary<NSString *, id<RLMBSON>> *> *)documents
+                 completion:(RLMMongoInsertManyBlock)completion {
+    [self collection:self.name].insert_many(static_cast<realm::bson::BsonArray>(RLMConvertRLMBSONToBson(documents)),
+                                            [completion](std::vector<realm::ObjectId> insertedIds,
+                                                         realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        NSMutableArray *insertedArr = [[NSMutableArray alloc]  init];
+        for (auto& objectId : insertedIds) {
+            [insertedArr addObject:[[RLMObjectId alloc] initWithValue:objectId]];
+        }
+        completion(insertedArr, nil);
+    });
+}
+
+- (void)aggregateWithPipeline:(NSArray<NSDictionary<NSString *, id<RLMBSON>> *> *)pipeline
+                   completion:(RLMMongoFindBlock)completion {
+    [self collection:self.name].aggregate(static_cast<realm::bson::BsonArray>(RLMConvertRLMBSONToBson(pipeline)),
+                                          [completion](realm::util::Optional<realm::bson::BsonArray> documents,
+                                                       realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(nil, RLMAppErrorToNSError(*error));
+        }
+        completion((NSArray<id> *)RLMConvertBsonToRLMBSON(*documents), nil);
+    });
+}
+
+- (void)countWhere:(NSDictionary<NSString *, id<RLMBSON>> *)document
+             limit:(NSInteger)limit
+        completion:(RLMMongoCountBlock)completion {
+    [self collection:self.name].count(static_cast<realm::bson::BsonDocument>(RLMConvertRLMBSONToBson(document)),
+                                      limit,
+                                      [completion](uint64_t count,
+                                                   realm::util::Optional<realm::app::AppError> error) {
+        if (error) {
+            return completion(0, RLMAppErrorToNSError(*error));
+        }
+        completion((int)count, nil);
```

Comment:
`int` is incorrect in all of these as there's platforms where it's 32 bits and NSInteger is 64.